pkg/k8s/pods/pod.go

```diff
@@ -444,3 +445,27 @@ func isRunning(p *apiv1.Pod) bool {
 
 	return false
 }
+
+func GetHealthcheckFailure(ctx context.Context, namespace, svcName, stackName string, c kubernetes.Interface) string {
```

Comment:
I would check if the service has `Endpoints` instead

Reply:
That is useful for knowing if the healthchecks have passed, however to know if what has failed is the healthcheck, the only option I can think of is checking events